### PR TITLE
Make Pimcore 6/Symfony 4.x compatible

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yeso
| BC breaks?    | no
| Deprecations? | yes
| Fixed tickets | #10

### CODE REVIEW

File: src/DynamicSearchBundle/DynamicSearchBundle.php [hunk 1]
```diff
@@ -68,7 +68,7 @@ public function getInstaller()
     public function getJsPaths()
     {
         return [
-            '/bundles/dynamicsearch/js/plugin.js',
```
Comment (dpfaffenbauer) [inline src/DynamicSearchBundle/DynamicSearchBundle.php]: file does not exist...

File: src/DynamicSearchBundle/Resources/config/services.yml [hunk 2]
```diff
@@ -8,6 +8,8 @@ services:
         autoconfigure: true
         public: false
 
+    DynamicSearchBundle\Service\LockServiceInterface: '@DynamicSearchBundle\Service\LockService'
```
Comment (dpfaffenbauer) [inline src/DynamicSearchBundle/Resources/config/services.yml]: it complained about autowiring on services that don't exist

File: composer.json [hunk 3]
```diff
@@ -25,10 +25,13 @@
             "bundles": [
                 "DynamicSearchBundle\\DynamicSearchBundle"
             ]
+        },
+        "branch-alias": {
+            "dev-master": "0.6-dev"
```
Comment (dpfaffenbauer) [inline composer.json]: always good to have a branch-alias ;)

File: src/DynamicSearchBundle/DependencyInjection/DynamicSearchExtension.php [hunk 4]
```diff
@@ -38,6 +40,13 @@ public function load(array $configs, ContainerBuilder $container)
             $container->addResource(new FileResource($providerConfig->locateConfigFile()));
         }
 
-        $container->set(ProviderConfig::class, $providerConfig);
+        $providerConfigDefinition = new Definition();
+        $providerConfigDefinition->setClass(ProviderConfig::class);
+
+        $container->setDefinition(ProviderConfig::class, $providerConfigDefinition);
+
+        $container
```
Comment (dpfaffenbauer) [inline src/DynamicSearchBundle/DependencyInjection/DynamicSearchExtension.php]: needs all of the tags here, tbd

File: src/DynamicSearchBundle/DependencyInjection/DynamicSearchExtension.php [hunk 5]
```diff
@@ -38,6 +40,13 @@ public function load(array $configs, ContainerBuilder $container)
             $container->addResource(new FileResource($providerConfig->locateConfigFile()));
         }
 
-        $container->set(ProviderConfig::class, $providerConfig);
```
Comment (dpfaffenbauer) [inline src/DynamicSearchBundle/DependencyInjection/DynamicSearchExtension.php]: that didn't work anymore... but it's ok to just have a separate service for it

File: src/DynamicSearchBundle/Resources/config/services.yml (hunk 2 above)
Comment (dpfaffenbauer) [reply src/DynamicSearchBundle/Resources/config/services.yml]: BTW: Public Bundles should not rely on autowiring: https://symfony.com/doc/current/service_container/autowiring.html#public-and-reusable-bundles

File: src/DynamicSearchBundle/Resources/config/services/logging.yml [hunk 6]
```diff
@@ -25,4 +30,4 @@ services:
     dynamic_search.log.formatter.console.provider:
         class: Symfony\Bridge\Monolog\Formatter\ConsoleFormatter
         arguments:
-            - "[%%datetime%%] %%level_name%%: %%start_tag%%%%message%%%%end_tag%% %%extra%%\n"
\ No newline at end of file
+            - '%dynamic_search.console.logger.options%'
```
Comment (solverat) [inline src/DynamicSearchBundle/Resources/config/services/logging.yml]: nice one!

**Comment by solverat** [comment]
thanks @dpfaffenbauer!

**Pull request merged**